src/detect-engine-file.c

```diff
@@ -194,7 +194,22 @@ uint8_t DetectFileInspectGeneric(DetectEngineCtx *de_ctx, DetectEngineThreadCtx
     if (ffc == NULL) {
         SCReturnInt(DETECT_ENGINE_INSPECT_SIG_CANT_MATCH_FILES);
     } else if (ffc->head == NULL) {
-        SCReturnInt(DETECT_ENGINE_INSPECT_SIG_NO_MATCH);
```

Comment:
When do we get `ffc != NULL && ffc->head == NULL` ?  

This still looks fishy to me as a signature with `file.data` that does not match today because of this `SCReturnInt(DETECT_ENGINE_INSPECT_SIG_NO_MATCH)`, will cause FP with this PR...

Reply:
Talking about fishy, I think it is a good catch ;) I'm going to add a suricata-verify test so we are sure this case is handled.

Reply:
Ok, but do you know, in plain English, when we have the cases :
- `ffc == NULL` (protocol or direction not supporting files ?)
- `ffc != NULL && ffc->head == NULL` protocol supporting files but tx does not have a file in this direction ? (that is what I expected but this seems a wrong expectation)